Added extension method for HttpResponseMessage

wosciatko opened the pull request:
Added extension method for HttpResponseMessage to create correct exception for response's status code.

Used this extension, where similar code was before

Formatted few long lines

Commit:
Added extensions method for HttpResponseMessage to create correct exception for response's status code.
And formatted some long lines

Commit:
Added unknown status code exception
Added RequestConflict excetpion to extension method
Little refactor of unit tests

Commit:
Empty space removed

satano commented on src/Kros.AspNetCore/Exceptions/UnknownStatusCodeException.cs:
```diff
@@ -0,0 +1,13 @@
+
+namespace Kros.AspNetCore.Exceptions
+{
+    class UnknownStatusCodeException : Exception
```
`public`

satano commented on tests/Kros.AspNetCore.Tests/Extensions/HttpResponseMessageExtensionsShould.cs:
```diff
@@ -0,0 +1,71 @@
+    public class HttpResponseMessageExtensionsShould
+    {
+        [Fact]
+        public void ReturnsForbideenExceptionForStatusCode403()
```
Prvých 5 testov sú úplne rovnaké, takže ich prerob na `Theory` - t. j. parametrický test, ktorý bude mať vstupy `HttpStatusCode` a výsledný typ výnimky.

Príklad: https://github.com/Kros-sk/Kros.AspNetCore/blob/master/tests/Kros.AspNetCore.Tests/HelpersShould.cs#L31

wosciatko replied:
Problem s Theory je, ze ju neviem spravit genericku, a parameter Type zas neviem (bez reflexie) pouzit ako generikum pre assertion

satano commented on tests/Kros.AspNetCore.Tests/Extensions/HttpResponseMessageExtensionsShould.cs:
```diff
@@ -0,0 +1,71 @@
+            var response = new HttpResponseMessage { StatusCode = HttpStatusCode.NoContent };
+            Exception ex = response.GetExceptionForStatusCode();
+            ex.Should().BeOfType<UnknownStatusCodeException>();
+            ex.Message.Should().Contain(((int) HttpStatusCode.NoContent).ToString());
```
Medzera za `(int)`.

Burgyn commented on src/Kros.AspNetCore/Authorization/GatewayAuthorizationMiddleware.cs:
```diff
@@ -89,7 +89,8 @@ internal class GatewayAuthorizationMiddleware
             {
                 client.DefaultRequestHeaders.Add(HeaderNames.Authorization, value.ToString());
 
-                HttpResponseMessage response = await client.GetAsync(_jwtAuthorizationOptions.AuthorizationUrl + httpContext.Request.Path.Value);
```
Ja som pôvodne myslel, že bude extension priamo nad clientom. Niečo v zmysle `GetAndCheckResponseAsync()` a ona skontroluje response a vyhodí rovno exception. Aby programátor nemusel potom písať už ani ten `if`. @satano  čo ty na to?

satano replied:
Dohodli sme sa, že ostane extension metóda nad `HttpResponseMessage` (napríklad `ThrowIfNotSuccessStatusCode()`), ktorá rovno spraví ten `if`. A zároveň sa spravia naše extension metódy nad klientom, ktoré budú používať tamto (`GetStringAndCheckResponse` - tu mám trochu problém s menom).

Burgyn replied:
Jop s tým menom mám problém aj ja :-) Dal som to len ako príklad, pretože nič lepšie mi nenapadlo.

wosciatko replied:
hotovo :)

Commit:
Added extensions method for HttpClient fot get with correct exception
Updated unit tests for HttpResponseMessage
Used new extension in GatewayAuthorizationMiddleware

Commit:
fixed empty line

Commit:
Added missing XML comment for HttpClientExtensions class
